lale/lib/aix360/logistic_rule_regression.py

```diff
@@ -0,0 +1,455 @@
+# Copyright  2020,2021 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import aix360.algorithms.rbm.beam_search
+import numpy as np
+import pandas as pd
+from aix360.algorithms.rbm import LogisticRuleRegression
+from sklearn.linear_model import LogisticRegression
+
+import lale
+import lale.docstrings
+import lale.operators
+from lale.lib.aif360.util import (
+    _categorical_input_predict_proba_schema,
+    _categorical_input_predict_schema,
+    _categorical_output_predict_proba_schema,
+    _categorical_output_predict_schema,
+    _categorical_supervised_input_fit_schema,
+)
+
+from .beam_search import beam_search, beam_search_K1
+
+aix360.algorithms.rbm.beam_search.beam_search = beam_search
+aix360.algorithms.rbm.beam_search.beam_search_K1 = beam_search_K1
+
+
+def add_method(cls):
+    """
+    Decorator function to add/replace methods to existing classes.
+    """
+
+    def decorator(func):
+        if type(cls) is list:
+            for c in cls:
+                setattr(c, func.__name__, func)
+        else:
+            setattr(cls, func.__name__, func)
+
+    return decorator
+
+
+@add_method(LogisticRuleRegression)
+def fit(self, X, y, Xstd=None):
```

Comment:
This method is almost identical to the corresponding [fit method](https://github.com/Trusted-AI/AIX360/blob/36459f2a585d0e2a2e8582562bf226d4402b57d6/aix360/algorithms/rbm/logistic_regression.py#L63-L208) in AIX360. The only difference appears to be four lines of code with variable initializations, namely:
```
12a13,15
>     numOrd = None
>     Astd = None
>     # B = None
162a166
>                 B = np.concatenate((Astd, A), axis=1)
```

Instead of copy-and-pasting this code, please remove it from this PR, and simply use the original fit method as-is.

Reply:
Hi  @hirzel  currently I am using the v0.1.0 branch of aix360(https://github.com/Trusted-AI/AIX360/tree/v0.1.0)  and not the latest branch. The latest(0.2.1) branch of AIX360 has some dependency issues. It supports hard dependency using tensorflow ==1.14 and it is not available for python 3.8 through pip dependencies. It causing  Lale build failure .
0.1.0 version does not have any such hard dependency for tensorflow but it has some bugs so we monkey patched above classes  from 0.2.1 version to 0.1.0 and now the build is passing and the code is also working fine.